Foundation/Array/Bitmap.hs

```diff
@@ -443,5 +446,23 @@ foldl' f initialAcc vec = loop 0 initialAcc
         | i .==# len = acc
         | otherwise  = loop (i+1) (f acc (unsafeIndex vec i))
 
+all :: (Bool -> Bool) -> Bitmap -> Bool
+all p bm = loop 0
```

Comment:
no, it's not yet optimised; it's still going to allocate through an integer. let's not bother about this here as it's not that critical; I'll look either into fixing the issue at the source of the problem, or by maybe doing some kind of global replace of 0 to azero later